vignettes/costofliving.Rmd

```diff
@@ -67,6 +76,8 @@ us_state_local_tax_rate
 
 ## Available Functions
 
+Note: The functions are able to handle syntax errors if the user inputs an invalid state abbreviation or duplicates of a state.
```

Comment:
How about we elaborate on this a bit? For eg give an example of what an invalid inut looks like? We don't have to run the code with the invalid input eg gross_salary("Mass") bc I don't think you can knit the vignette with erroneous code

Reply:
Note: The functions are able to handle syntax errors if the user inputs an invalid state abbreviation (for example, `gross_salary("ma")` or `gross_salary("Massachusetts")`  instead** of `gross_salary("MA")`), or duplicates of a state (for example, `avg_col_multiple_states(c("MA", "TX", "MA"))`).

Thinking of something like this, but idk how to best structure it yet

Reply:
What do you think of this?
Note: The functions are able to handle syntax errors if the user inputs a invalid state abbreviation or duplicates of a state. For example, gross_salary("ma") or gross_salary("Massachusetts") will throw an error that the input must be a valid state abbreviation and the letters are capitalized. gross_salary("MA") is the correct way to use the function. Duplicate inputs are also invalid, such as avg_col_multiple_states(c("MA", "TX", "MA")) and can be corrected as avg_col_multiple_states(c("MA", "TX")).

Reply:
Note: The functions are able to handle syntax errors if the user inputs an invalid state abbreviation or duplicates of a state. For example, `gross_salary("ma")` or `gross_salary("Massachusetts")` will throw an error that the input must be a valid state abbreviation, with all letters being capitalized, and `gross_salary("MA")` is the correct way to use the function. For functions that require a vector of states as input, duplicate states in the input vector are also invalid, such as `avg_col_multiple_states(c("MA", "TX", "MA"))`, and the correct use should be `avg_col_multiple_states(c("MA", "TX"))`.

How about this? In general I think it sounds good, just rephrasing it a bit based on what I think but lmk if you have different opinions on this

Reply:
Note: The functions are able to handle syntax errors if the user inputs an invalid state abbreviation or duplicates of a state. For example, gross_salary("ma") or gross_salary("Massachusetts") will throw an error that the input must be a valid state abbreviation and the letters are capitalized. Thus, gross_salary("MA") is the correct way to use the function. Duplicate states are also invalid, such as avg_col_multiple_states(c("MA", "TX", "MA")) and can be corrected as avg_col_multiple_states(c("MA", "TX")).

Reply:
Note: The functions are able to handle syntax errors if the user inputs an invalid state abbreviation or duplicates of a state. For example, gross_salary("ma") or gross_salary("Massachusetts") will throw an error that the input must be a valid state abbreviation and the letters are capitalized. Thus, gross_salary("MA") is the correct way to use the function. For functions that require a vector of states as input, duplicate states in the input vector are also invalid, such as avg_col_multiple_states(c("MA", "TX", "MA")), and the correct use should be avg_col_multiple_states(c("MA", "TX")).

Reply:
> For example, gross_salary("ma") or gross_salary("Massachusetts") will throw an error that the input must be a valid state abbreviation and the letters are capitalized. Thus, gross_salary("MA") is the correct way to use the function. 

I find the link between these two sentences to be a bit... awkward? If you want, you can also keep it as before, without the "Thus, "

> Duplicate states are also invalid, such as avg_col_multiple_states(c("MA", "TX", "MA")) and can be corrected as avg_col_multiple_states(c("MA", "TX")).

I think it's better to specify that this only applies to functions that require a vector of states to be the input. Also, the "can be corrected as" part sounds not really smooth to me (sorry maybe I'm just nitpicking lol). How about
 
>  For functions that require a vector of states as input, duplicate states are also invalid, such as avg_col_multiple_states(c("MA", "TX", "MA")). The correct use should be `avg_col_multiple_states(c("MA", "TX"))`.

Reply:
Maybe this one?

Note: The functions are able to handle syntax errors if the user inputs an invalid state abbreviation or duplicates of a state. For example, gross_salary("ma") or gross_salary("Massachusetts") will throw an error that the input must be a valid state abbreviation and the abbreviation must be capitalized. Thus, gross_salary("MA") is the correct way to use the function. For functions that require a vector of states as input, duplicate states in the input vector are also invalid, such as avg_col_multiple_states(c("MA", "TX", "MA")), and can be corrected to avg_col_multiple_states(c("MA", "TX")).

Reply:
I just thought the sentence was getting too long which is why I broke it up.

Reply:
I agree to add that we should add vector of states as input

Reply:
Note: The functions are able to handle syntax errors if the user inputs an invalid state abbreviation or duplicates of a state. For example, gross_salary("ma") or gross_salary("Massachusetts") will throw an error that the input must be a valid state abbreviation and the letters are capitalized. gross_salary("MA") is the correct way to use the function. For functions that require a vector of states as input, duplicate states are also invalid, such as avg_col_multiple_states(c("MA", "TX", "MA")). The correct use should be avg_col_multiple_states(c("MA", "TX")).

Reply:
hopefully this is the last one because it sounds good to me?

Reply:
Yeah it sounds good to me too

Just don't forget the `` for code snippets